manifests/config.pp

```diff
@@ -238,4 +239,47 @@
     }
   }
 
+  file { dirname($pid):
+    ensure  => directory,
+    owner   => $setuid,
+    group   => $setgid,
+    mode    => '0644',
+    seluser => 'system_u',
+    selrole => 'object_r',
+    seltype => 'stunnel_var_run_t',
+  }
+
+
+  # These templates need variables, that's why they are here
+  if 'systemd' in $facts['init_systems'] {
+    file { '/etc/systemd/system/stunnel.service':
+      ensure  => file,
+      content => template('stunnel/connection_systemd.erb'),
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0644',
+      notify  => Exec['stunnel daemon reload']
+    }
+    exec { 'stunnel daemon reload':
+      command     => '/usr/bin/systemctl daemon-reload',
+      refreshonly => true,
+    }
+
+  }
+  else {
+    file { '/etc/rc.d/init.d/stunnel':
+      ensure  => 'present',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0750',
+      content => template("stunnel/connection_init.erb"),
+      notify  => Exec['stunnel_chkconfig_update']
+    }
+    exec { 'stunnel_chkconfig_update':
+      command     => '/sbin/chkconfig --del stunnel; /sbin/chkconfig --add stunnel',
```

Comment:
I don't think this is needed, as we are not changing the run level or start/stop priorities.